hasktorch/src/Torch/Typed/NN/Convolution.hs

```diff
@@ -84,10 +91,16 @@ instance
        ],
     ConvSideCheck inputSize kernelSize stride padding outputSize
   ) =>
-  HasForward (Conv1d inputChannelSize outputChannelSize kernelSize dtype device) (Tensor device dtype '[batchSize, inputChannelSize, inputSize], Proxy stride, Proxy padding) (Tensor device dtype '[batchSize, outputChannelSize, outputSize])
+  HasForward
+    (Conv1d inputChannelSize outputChannelSize kernelSize dtype device)
+    (Tensor device dtype '[batchSize, inputChannelSize, inputSize], Proxy stride, Proxy padding)
   where
+  type
+    Output
+      (Conv1d inputChannelSize outputChannelSize kernelSize dtype device)
+      (Tensor device dtype '[batchSize, inputChannelSize, inputSize], Proxy stride, Proxy padding) =
+      Tensor device dtype '[batchSize, outputChannelSize, outputSize]
```

Comment:
This fails to compile because `outputSize` is unknown here. This worries me a bit that we have cases where the output type is polymorphic and only reified in context; so it's not possible in general to replace the type parameter `b` with a type family. But I might be missing something.

Reply:
Sorry, that was a confusing comment, and possibly misused some overloaded terminology; my essential point of confusion is this: what happens when the output type (e.g. the dimensions of the output tensor) are polymorphic (and therefore context-dependent)? For example, in `Conv1d` before this PR, `forward` is polymorphic in `inputSize` and `outputSize`, and we use the constraint family `ConvSideCheck` to constrain to valid dimensions. But in this case it seems like there might be many possible output shapes consistent with the input shape (this is where I think I'm wrong... still need to understand `ConvSideCheck`). In any case, it seems like we'd need to rewrite the constraint into a type family from input to output. I'll take a look at this next.

Reply:
Resolved in Slack conversation! I don't think my concern above is valid; there should not be any cases where `forward` is polymorphic in the output type (e.g. allowing different shapes depending on context)